config/socket/game.js

```diff
@@ -1,428 +1,577 @@
     }
-  });
-  return playerIndex;
-};
+    return cards;
+  }
+
+  /**
+   * 
+   * @param {*} maxAnswers 
+   * @returns {*} dealAnswers
+   */
+  dealAnswers(maxAnswers) {
+    maxAnswers = maxAnswers || 10;
+    const storeAnswers = (err, data) => {
+      this.answers = data;
+    };
+    for (let i = 0; i < this.players.length; i++) {
+      while (this.players[i].hand.length < maxAnswers) {
+        this.players[i].hand.push(this.answers.pop());
+        if (!this.answers.length) {
+          this.getAnswers(storeAnswers);
+        }
+      }
+    }
+  }
+   
+  /**
+   * 
+   * @param {*} thisPlayer
+   * @returns {*} _findPlayerIndexBySocket
+   */
+  _findPlayerIndexBySocket(thisPlayer) {
+    let playerIndex = -1;
+    _.each(this.players, (player, index) => {
+      if (player.socket.id === thisPlayer) {
+        playerIndex = index;
+      }
+    });
+    return playerIndex;
+  }
 
-Game.prototype.pickCards = function(thisCardArray, thisPlayer) {
-  // Only accept cards when we expect players to pick a card
-  if (this.state === "waiting for players to pick") {
+  /**
+   * 
+   * @param {*} thisCardArray
+   * @param {*} thisPlayer
+   * @returns {*} pickCards
+   */
+  pickCards(thisCardArray, thisPlayer) {
+    // Only accept cards when we expect players to pick a card
+    if (this.state === 'waiting for players to pick') {
     // Find the player's position in the players array
-    var playerIndex = this._findPlayerIndexBySocket(thisPlayer);
-    console.log('player is at index',playerIndex);
-    if (playerIndex !== -1) {
+      let playerIndex = this._findPlayerIndexBySocket(thisPlayer);
+      console.log('player is at index', playerIndex);
+      if (playerIndex !== -1) {
       // Verify that the player hasn't previously picked a card
-      var previouslySubmitted = false;
-      _.each(this.table, function(pickedSet, index) {
-        if (pickedSet.player === thisPlayer) {
-          previouslySubmitted = true;
-        }
-      });
-      if (!previouslySubmitted) {
-        // Find the indices of the cards in the player's hand (given the card ids)
-        var tableCard = [];
-        for (var i = 0; i < thisCardArray.length; i++ ) {
-          var cardIndex = null;
-          for (var j = 0; j < this.players[playerIndex].hand.length; j++) {
-            if (this.players[playerIndex].hand[j].id === thisCardArray[i]) {
-              cardIndex = j;
+        let previouslySubmitted = false;
+        _.each(this.table, (pickedSet, index) => {
+          if (pickedSet.player === thisPlayer) {
+            previouslySubmitted = true;
+          }
+        });
+        if (!previouslySubmitted) {
+        // Find the indices of the cards in the player's
+        // hand (given the card ids)
+          const tableCard = [];
+          for (let i = 0; i < thisCardArray.length; i += 1) {
+            let cardIndex = null;
+            for (let j = 0; j < this.players[playerIndex].hand.length; j += 1) {
```

Comment:
and here too

Reply:
Due changes will be effected immediately